src/modules/simulator/simulator.h

```diff
@@ -180,6 +182,8 @@ class Simulator : public ModuleParams
 
 	InternetProtocol _ip{InternetProtocol::UDP};
 
+	char *_ipaddr;
```

Comment:
What about using this field to signal whether TCP is remote or not?

So, I would suggest to rename it and set it to `nullptr` by default:

```suggestion
	char *_tcp_remote_ipaddr{nullptr};
```